packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -254,7 +264,26 @@ class LinkTriggerSignals {
     _hasDomEvent = true;
     _viewIdFromDomEvent = viewId;
     _mouseEvent = mouseEvent;
-    _didUpdate();
+    _scheduleReset();
+
+    if (mouseEvent != null) {
+      canBrowserNavigate = true;
+      // We have to make a decision whether to allow the browser to navigate or not. The decision
+      // has to be made before the end of the event loop.
+      //
+      // If we don't hear back from the `followLink` signal by the end of the event loop, we should
+      // prevent the browser from navigating. If the `followLink` signal arrives later, we can
+      // navigate programmatically using `launchUrl`.
+      scheduleMicrotask(() {
+        if (!_hasDomEvent) {
+          // By the time the microtask runs, the link may have already been triggered. In that case,
+          // we want to do nothing here.
+          return;
+        }
+        canBrowserNavigate = false;
+        mouseEvent.preventDefault();
+      });
```

Comment:
Instead of calling `_scheduleReset()`, should we just call `reset` directly at the beginning of this microtask?

In this case we really don't have to wait a "long" timeout to reset the state of the plugin, do we? Is it because of how `followLink` might happen at a random time?

Reply:
Yeah, we have to wait for `followLink`, which may (or may not) come later. So the timer is really acting like a deadline for waiting for `followLink`. If `followLink` doesn't show up by that time, we assume it's never coming.

In the normal case, `followLink` should arrive before the end of the microtask and everything works beautifully. But in semantics mode, we have a thing called "click debouncer" that delays pointer events. That's what causes the framework's `onTap` callback to be fired late.